### PR TITLE
Functional programming 👉 partial functions

### PR DESCRIPTION
**Pull request by pmcnr-hx**
Some of us are partial (pun intended) to the functional programming paradigm. This can be a bit confusing for those who have less exposure to the _functional_ way of doing things. We don't claim that _functional_ is better than _imperative_ (although it mostly is) but try instead to clarify the functional alternatives to the more traditional constructs.
- [x] :+1: 
- [x] :+1: :+1:

### CODE REVIEW

File: serverside-javascript-best-practices.md [hunk 1]
```diff
@@ -680,6 +680,65 @@ async.map(['http://foo', 'http://bar'], dbInfo, ...);
...
+```
+
+The following are all equivalent ways to implement the `subtractFrom5` and
+`subtract5` functions using `subtract`:
+
+* Wrapper functions:
+
+  ```javascript
+var subtractFrom5 = function (x) {
+  return subtract(5, x);
```
Comment (hx-markterry) [inline serverside-javascript-best-practices.md]: There is a problem with the indentation here, when the markdown is viewed on github.

**Comment by hx-markterry** [comment]
I agree with the above, but it should be moved to the general javascript best practices document as it is not server specific.

**Comment by pmcnr-hx** [comment]
@hx-markterry: I've added here because @hxoliverrumbelow mentioned we would be merging the general javascript best practices document with this one. Will we actually be doing that sometime soon or will these always be separate documents and we'll just move part of the server-side best practices that apply generally to the general document?

**Comment by hx-markterry** [comment]
I'm sure there will be some server or client specific parts of the javascript best practices, how it is arranged does not really matter as long as it is clear. My main concern here is that the best practice suggested above applies to both and should be under the general document, rather than deferring the merging of practices to someone else or another point in time.

**Comment by pmcnr-hx** [comment]
Will move to general @hx-markterry.

**Comment by pmcnr-hx** [comment]
@hx-markterry: moved to general best practices.

**Comment by hx-markterry** [comment]
ok with me :monkey:

**Comment by pmcnr-hx** [comment]
This has now been rebased on current master.

**Comment by philmeehan** [comment]
:+1:

**Pull request merged**